### DISCUSSION TITLE
Render HTML into native view

### DISCUSSION THREAD

**Discussion by darko-bw** [NativeScript/rfcs#18] [1. Ideas & Discussions]
ReactNative has got an awesome plugin to render simple HTML into native view:

https://github.com/archriss/react-native-render-html

Is there any possibility to get any thing similar in NS8? I guess we still can use HtmlView but wanted to have something more advanced.

**Comment by 7ammer** [comment]
I really like this feature "Custom HTML tags": https://github.com/meliorence/react-native-render-html#creating-custom-renderers

**Comment by 7ammer** [comment]
I guess to achieve this you would need some sort of html parser (ie: https://www.npmjs.com/package/node-html-parser) and then a way to map each html tag to a nativescript component.

**Comment by NathanaelA** [comment]
If I understand what you are asking for:
`<BlueLabel text="blah"/>`  to automatically behind the scenes create a label does something like `<Label class="blueStyling" text="blah"/>`
In NS-Core, you can register new elements which do this behind the scenes.   In Angular & Vue they both have a extension type that allows you to create a component that map to do special things. 

 For example in one of my Vue apps code, I put this:
`<IconButton row="1" icon="&#xf24d;" text="TEST BUTTON" iconset="fas" @tap="tap" margin="5 10" border="true"/>`

And it creates this: 
![image](https://user-images.githubusercontent.com/850871/109370603-9d8a0e80-7866-11eb-88dd-705a401b81ee.png)
This is NOT a plugin, this is just a component I created using normal NS Layout code, so that I can layout multiple of these buttons with different icons, colors, text in a app I was working on.

So you could easily register DIV, SPAN, HR, etc to return native components with the flavor you are using...

**Reply by 7ammer** [reply]
Hi @NathanaelA,

I think the OP is looking for a better nativescript `<htmlView>` element (https://docs.nativescript.org/ui/components/html-view). A bit like the react-native-render-html plugin.

I mentioned the "custom html tags" feature because it allows the user to fully customise how a particular html tag should be rendered.
For example:
`<h1>Hello world</h1>` could be mapped to `<Label class="h1" text="text.value" />`.
`<img src="my-image.jpg" />` could be mapped to `<MyImageCacheComponent ratio="16:9" src="src.value" />`.

I don't think this is particularly hard to do if I ever needed to build something like this, but it would be convenient to have this incorporated into `<htmlView>` or a similarly named component.

For me personally, `<htmlView>` has never been customisable enough for me to ever be able to use it in a project.
